docs/x509.rst

```diff
@@ -652,6 +652,32 @@ X.509 Extensions
     purposes indicated in the key usage extension. The object is
     iterable to obtain the list of :ref:`extended key usage OIDs <eku_oids>`.
 
+.. class:: AuthorityKeyIdentifier
+
+    .. versionadded:: 0.9
+
+    The authority key identifier extension provides a means of identifying the
+    public key corresponding to the private key used to sign a certificate.
+
+    .. attribute:: key_identifier
+
+        :type: bytes
+
+        A value derived from the public key used to verify the certificate's
+        signature.
```

Comment:
... wtf RFC 5280. Can we get some explicit language about how the contents
of this is allowed to verify or something.

On Sat, Apr 18, 2015 at 10:06 PM, Paul Kehrer <notifications@github.com>
wrote:

> In docs/x509.rst
> <https://github.com/pyca/cryptography/pull/1835#discussion_r28649105>:
>
> > @@ -652,6 +652,32 @@ X.509 Extensions
> >      purposes indicated in the key usage extension. The object is
> >      iterable to obtain the list of :ref:`extended key usage OIDs <eku_oids>`.
> >
> > +.. class:: AuthorityKeyIdentifier
> > +
> > +    .. versionadded:: 0.9
> > +
> > +    The authority key identifier extension provides a means of identifying the
> > +    public key corresponding to the private key used to sign a certificate.
> > +
> > +    .. attribute:: key_identifier
> > +
> > +        :type: bytes
> > +
> > +        A value derived from the public key used to verify the certificate's
> > +        signature.
>
> The spec is noncommittal on what that means, unfortunately. There's a
> "standard" way that is pretty much what everybody does (to my knowledge)
> but technically you can put whatever the hell you want here. RFC 5280 is
> not great sometimes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1835/files#r28649105>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084